### PR TITLE
Side Bar Functionality

### PR DESCRIPTION
**Pull request by Nervith**
This makes the sidebar, and the basics of the logged in function

### CODE REVIEW

File: .idea/Untitled.iml
Comment (Waluiyeegi) [inline .idea/Untitled.iml:1]: When pushing try not to push any of idea files, so just push the files that you changed and leave the idea files on your local device.

File: .idea/modules.xml
Comment (Waluiyeegi) [inline .idea/modules.xml:1]: same as above

File: server/target/classes/Main.class
Comment (Waluiyeegi) [inline server/target/classes/Main.class:1]: i think we dont wanna push anything in the target files as well

File: client/src/App.svelte [hunk 1]
```diff
@@ -1,9 +1,115 @@
 <script>
```
Comment (Zhangjason308) [inline client/src/App.svelte:1]: You shouldn't mutate the App.Svelte file
We use a router

File: client/src/components/Dashboard.svelte [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<script>
```
Comment (Zhangjason308) [inline client/src/components/Dashboard.svelte:1]: shouldn't need this dashboard

File: server/target/classes/Main.class
Comment (Zhangjason308) [reply server/target/classes/Main.class:1]: yeah please undo the unneeded file pushes like /target and /idea

**Review by Zhangjason308 — CHANGES_REQUESTED** [review]
Merge main into your branch. Its currently outdated since we add alot of new changes

File: .idea/modules.xml
Comment (nicholasyung1) [reply .idea/modules.xml:1]: Agreed, should only be submitting .Java, .JS, .Svelte files

File: client/src/components/MainLayout.svelte [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<script>
+    import Sidebar from './Sidebar.svelte';
+    import { Router, Route } from 'svelte-routing';
+    import Home from './Home.svelte';
+    import Login from './Login.svelte';
+    import Register from './Register.svelte';
+    import Profile from './Profile.svelte';
+
+    let isLoggedIn = true; // Simulated login state
```
Comment (nicholasyung1) [inline client/src/components/MainLayout.svelte:9]: Should remove testing materials when merging to main.

File: client/src/components/MainLayout.svelte [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        isLoggedIn = false;
+    }
+</script>
+
+<div class="main-layout">
+    <Sidebar {isLoggedIn} />
+
+    <main class="router-container">
+        <Router>
+            <Route path="/" component={Home} />
```
Comment (nicholasyung1) [inline client/src/components/MainLayout.svelte:25]: I think this already exists in router.Svelte file.

@Zhangjason308, please confirm

**Pull request closed without merging**